icefoganalytics/travel-authorization
# Implement Correcting Lines for Non-Travel Status Days on Estimate Tab

Fixes https://github.com/icefoganalytics/travel-authorization/issues/119

Relates to:
- https://github.com/icefoganalytics/travel-authorization/issues/68

# Context

When a user files a travel authorization request, they have the ability to enter "non-travel status days" in the form. This feature needs to affect the estimate generation system. 

Where the user enters non-travel status days of "4: on the estimate tab, we need to add a couple rows - I'm not sure how many - that are correcting lines. The will subtracts the cost of 4 days of accommodations and 4 days worth of per-diems. 

In the travel authorization form we don't track which days are non-travel status days, so we can't remove the exact days.

# Implementation

1. Implement correcting entry for non-travel status days.
2. Switch to using cleaner interface for transient/association factory params.

# Screenshots

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/fd7cee41-ca94-485b-a655-ca42525be4fe)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
3. Boot the app via `dev up`
4. Log in to the app at http://localhost:8080
5. Go to the "My Travel Requests" page via the top drop down nav.
6. Create a new "Travel Authorization" via the top right button.
7. Go to the "Details" section.
8. Make a trip from Whitehorse to Vancouver on the 5th of the current month.
9. Don't set a time for the departure date from Whitehorse.
10. Leave the travel method and accommodation type as their defaults of Aircraft + Hotel.
11. Make the return date the 7th of the month at 15:00 (3pm).
12. Set the "Days on non-travel status" to 4. Even though this is a 3 day trip.
13. Go to the "Approvals" Section.
14. Generate the estimate via the call-to-action button.
15. Click the "Edit Estimate" button to view the generated estimates.
16. Check that the estimates calculations for per-diems match expected values calculated in BC (Rest of Canada).
    1. Check that the "Non-Travel Status" entry looks like amount  `-$775.55`
        > 3 day @ non-travel status per diem -275.55 and 2 day @ non-travel status accommodation -500.00
    2. Note that the max non-travel status entries are the max number of related entries that exist. 3 for means, 2 for accommodations
